src/foam/nanos/theme/SubdomainAwareDAO.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.theme',
+  name: 'SubdomainAwareDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: 'DAO that adds value to subdomains property',
+
+  javaImports: [
+    'javax.servlet.http.HttpServletRequest'
+  ],
+
+  methods: [
+    {
+      name: 'put_',
+      javaCode: `
+        if ( obj instanceof SubdomainAware ) {
+          if ( "".equals(((SubdomainAware) obj).getSubdomain()) ) {
```

Comment:
Can just use the isEmpty method on Java Strings so would be something like 

`((SubdomainAware) obj).getSubdomain()).isEmpty()`